pkg/chunkenc/unordered.go

```diff
@@ -88,14 +90,14 @@ func (hb *unorderedHeadBlock) UncompressedSize() int {
 	return hb.size
 }
 
-func (hb *unorderedHeadBlock) Reset() {
-	x := newUnorderedHeadBlock(hb.format)
+func (hb *unorderedHeadBlock) Reset(symbolizer *symbolizer) {
```

Comment:
Looks like, the same symbolizer is reused after calling `Reset`.
https://github.com/grafana/loki/blob/238b174416be3d8d515722527a2802b6abfd9fbc/pkg/chunkenc/memchunk.go#L905

To minimize passing the symbolizer around, I think it makes more sense to remove the symbolizer argument and just pass `hb.symbolizer` to `newUnorderedHeadBlock`